sdks/go/pkg/beam/core/typex/special.go

```diff
@@ -64,6 +65,23 @@ type Window interface {
 	Equals(o Window) bool
 }
 
+type Timing string
+
+const (
+	EARLY   Timing = "EARLY"
+	ON_TIME Timing = "ON_TIME"
+	LATE    Timing = "LATE"
+	UNKNOWN Timing = "UNKNOWN"
+)
+
+type PaneInfo struct {
+	Timing              Timing
+	IsFirst             bool
+	IsLast              bool
+	Index               int64
+	NonSpeculativeIndex int64
```

Comment:
Note, there's nothing wrong with the current approach, I'm pointing out a thing about Go:

In this case, you can write these fields more compactly as: 
```suggestion
	IsFirst, IsLast				bool
	Index, NonSpeculativeIndex		int64
```